release.config.js

```diff
@@ -0,0 +1,118 @@
+// parser opts ensure that breaking changes indicated with `!` after the scope (before `:`)
+// are valid commits and are processed correctly.
+let parserOpts = {
+    "headerPattern": /^(\w*)(?:\(([\w\$\.\-\* ]*)\))?(!?)\: (.*)$/,
+    "headerCorrespondence": ["type", "scope", "breaking", "subject"],
+    "noteKeywords": ["BREAKING CHANGE", "BREAKING-CHANGE"]
+}
+
+// commit metadata transform based on writer opts from @conventional-changelog/conventional-changelog-angular
+// https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-angular/writerOpts.js
+// Adapted for our convention.
+let releaseNoteTransform = (commit, context) => {
+    console.log(commit)
+    let discard = true
+    const issues = []
+
+    commit.notes.forEach(note => {
+        note.title = 'BREAKING CHANGES'
+        discard = false
+    })
+
+    if (commit.type === 'feat') {
+        commit.type = 'Features'
+    } else if (commit.type === 'fix') {
+        commit.type = 'Bug Fixes'
+    } else if (discard) {
+        return
```

Comment:
As discussed, instead of discarding commits that don't match the conventional commit, lets introduce an "Other" section and add them there. We can remove the section when we are sure that it is always empty (i.e. commitlint is doing its intended job)

Reply:
And even better, when there’s no need for a section, it just doesn’t show up.

The fallback would still be there.

Reply:
I would focus on the types of changes that matter to our consumers:
- all commits that change the artefacts we are shipping, since those are potential sources of bugs / behavioural changes (`feat`, `fix`, `refactor`, `dep`)
- documentation changes (`doc`); with a bosh release these are also part of the build output, so they also qualify for the first bullet point

`test` and `ci` do not (or at least should not) affect the build output and therefore shouldn't affect our consumers. As such I don't see the need to include them in the release notes.

Reply:
I fully agree with that. There are now two separate lists of commit type to heading mapping, one for "changes" that always show up in the release note, and one for "hidden" commits that only show up if they have a breaking change.

Mapping is exactly as you laid out.

Any commits not fitting into that schema will show up in "Other", as discussed.